eng/pipelines/templates/steps/upload-repository-on-failure.yml

```diff
@@ -3,3 +3,12 @@ steps:
   condition: or(eq(variables['CaptureRepositoryOnFailure'], 'true'), and(failed(), eq(variables['Build.Reason'],'Schedule')))
   displayName: 'Capture repo state for analysis'
   artifact: repository-$(System.JobName)
+
+- pwsh: |
+    # Remove any installed packages that we build in the repo so we don't cache temporary builds
+    Remove-Item "$(MAVEN_CACHE_FOLDER)/com/azure" -Recurse -ErrorAction Ignore
```

Comment:
@alzimmermsft do you think removing just those 2 directories is good enough to clean the cache?

Reply:
Yes, those directories should contain all `groupId`s that are in use for Java Azure SDKs